realm/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -402,70 +402,70 @@ public TableQuery notEqualTo(long columnIndex[], String value){
     }
     protected native void nativeNotEqual(long nativeQueryPtr, long columnIndex[], String value, boolean caseSensitive);
 
-    public TableQuery beginsWith(long columnIndex, String value, boolean caseSensitive){
+    public TableQuery beginsWith(long columnIndices[], String value, boolean caseSensitive) {
```

Comment:
I would tend to agree, even though this class is not public facing, it still generates a lot of ugly boilerplate elsewhere.